### DISCUSSION TITLE
Welcome to edubeam Discussions!

### DISCUSSION THREAD

**Discussion by janvorisek** [janvorisek/edubeam#238] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by HienLe1198** [comment]
I want to create a function to calculate Mmax, Qmax, and draw an internal force diagram like the solution shown in the picture.
However, it is quite difficult because I need to determine the problem structure first (how many elements, nodes, supports, etc.), and each configuration requires a different solution.
Can you give me some hints on how to do this?

**Reply by janvorisek** [reply]
> I want to create a function to calculate Mmax, Qmax, and draw an internal force diagram like the solution shown in the picture.
> However, it is quite difficult because I need to determine the problem structure first (how many elements, nodes, supports, etc.), and each configuration requires a different solution.
> Can you give me some hints on how to do this?

@HienLe1198 
You need nodes at the point of interest (supports, loads, etc.). In your case the nodes are given - A, B, C and D.
I've modelled your structure in edubeam. See [here](https://run.edubeam.app/?model=eyJuIjpbWyJBIixbMCwwLDBdLFtdLG51bGxdLFsiQyIsWzUsMCwwXSxbXSxudWxsXSxbIkIiLFsyLDAsMF0sWzAsMl0sbnVsbF0sWyJEIixbNiwwLDBdLFsyXSxudWxsXV0sImUiOltbIjEiLFsiQSIsIkIiXSwiMSIsIjEiLFtmYWxzZSxmYWxzZV1dLFsiMiIsWyJCIiwiQyJdLCIxIiwiMSIsW2ZhbHNlLGZhbHNlXV0sWyIzIixbIkMiLCJEIl0sIjEiLCIxIixbZmFsc2UsZmFsc2VdXV0sIm0iOltbIjEiLDQwMDAsMjEwMDAwMDAwMDAwLDg3NTAwMDAwMDAwLDAuMDAwMDEyXV0sImNzIjpbWyIxIiwxLDAuMDAwMDgzNTYsMSwxZSszMl1dLCJlbCI6W1siMSIsWzAsNTAwMF0sdHJ1ZV1dLCJubCI6W1siQyIseyIwIjowLCIyIjoxMDAwMCwiNCI6MH1dXSwiZCI6W119). We don't compute the position of the extrema yet, but you can calculate that from the Q(x) function.

**Reply by HienLe1198** [reply]
Thank you for your reply. If I want to create a triangular or trapezoidal load, how can I do that and recalculate the functions that generate the shear force, normal force, and other results? Could you point me to the file that contains the function I should modify?

**Reply by janvorisek** [reply]
> Thank you for your reply. If I want to create a triangular or trapezoidal load, how can I do that and recalculate the functions that generate the shear force, normal force, and other results? Could you point me to the file that contains the function I should modify?

Unfortunately, we do not support triangular loads at the moment. It would have to be implement first into [this library](https://github.com/janvorisek/ts-fem/) and then handled correctly in edubeam user interface and its import/export/sharing.

I will try to implement this during the winter holidays. There are some more minor improvements planned, but I'm short on time at the moment.

**Reply by janvorisek** [reply]
@HienLe1198 Hi! Good news - linear element loads are now implemented. You can model triangular/trapezoidal loads and generate the corresponding internal force results.
